chttp/cookieauth.go

```diff
@@ -89,7 +89,19 @@ func (a *CookieAuth) RoundTrip(req *http.Request) (*http.Response, error) {
 	if err := a.authenticate(req); err != nil {
 		return nil, err
 	}
-	return a.transport.RoundTrip(req)
+
+	res, err := a.transport.RoundTrip(req)
+	if err != nil {
+		return res, err
+	}
+
+	if res != nil && res.StatusCode == http.StatusUnauthorized {
+		if a.Cookie() != nil {
+			a.client.Jar = nil
+			a.setCookieJar()
```

Comment:
Nuking the entire cookie jar feels a bit heavy-handed. On the other hand, I don't see any obvoius way to delete a specific cookie.  :thinking:   Perhaps setting it to a value that's already expired would work.

In general, it probably never matters, but I suppose it's possible that with the use of proxy servers or URL redirection, there might be some other cookies stored in there, too, in some cases...

Reply:
I agree, I can experiment with deleting the single cookie as it probably is possible as you suggested.

Reply:
This should be dealt with now.